alpine linux build errors

kazimsarikaya opened the pull request:
The closed issue  https://github.com/389ds/389-ds-base/issues/3394 didn't help to compile at alpine linux.

As I mention at that closed issue i commit several changes to compile successfully at alpine linux based on the patch https://github.com/389ds/389-ds-base/issues/3394#issuecomment-691626102

mreynolds389 committed:
Issue 4262 - Remove legacy tools subpackage

Description:  Remove all the legacy tool scripts, libraries, and obsolete files

Relates: https://github.com/389ds/389-ds-base/issues/4262

Reviewed by: viktor & firstyear (Thanks!!)

Apply Viktor's suggestions

kazimsarikaya committed:
alpine linux build

kazimsarikaya committed:
alpine linux build

kazimsarikaya committed:
alpine linux build

kazimsarikaya committed:
Merge branch 'master' of https://github.com/kazimsarikaya/389-ds-base

mreynolds389 committed:
Issue 2526 - revert API change in slapi_be_getsuffix()

Description:  The public slapi API was changed in the previous commit and
              this broke several other projects using 389 DS.  This patch
              restore the API, but it ignores the unused value:

                  const Slapi_DN *slapi_be_getsuffix(Slapi_Backend *be, int unused);

Relates: https://github.com/389ds/389-ds-base/issues/2526

Reviewed by: mreynolds

vashirov commented on ldap/servers/plugins/sync/sync_persist.c:
```diff
@@ -6,8 +6,8 @@
 
-#include <nspr4/prlog.h>
-#include <bits/stdint-intn.h>
+#include <nspr/prlog.h>
```
I think this can be just ```#include <prlog.h>```

kazimsarikaya replied:
it seems ok, too. also the prlog.h is not required at that file.

kazimsarikaya replied:
i deleted include

vashirov commented on ldap/servers/plugins/sync/sync_persist.c:
```diff
@@ -6,8 +6,8 @@
-#include <nspr4/prlog.h>
-#include <bits/stdint-intn.h>
+#include <nspr/prlog.h>
+#include <bits/stdint.h>
```
This is not present on glibc-based system. Perhaps it should be ifdefed?

kazimsarikaya replied:
i also tested, **bits/stdint-intn.h** is not neccessary. i will try build at centos and ubuntu to verift not to broke anything.

then i will update pr.

the header **bits/stdint-intn.h** is very interesting. it's at ubuntu as **/usr/include/x86_64-linux-gnu/bits/stdint-intn.h** and referenced from **/usr/include/stdint.h**

kazimsarikaya replied:
i deleted include

kazimsarikaya committed:
Merge branch 'master' of https://github.com/kazimsarikaya/389-ds-base

kazimsarikaya committed:
delete unnecessary includes

kazimsarikaya committed:
delete unnecessary includes

mreynolds389 committed:
Issue 4262 - Remove legacy tools subpackage (restart instances after rpm install)

Description:  Update specfile to restart instances after installing new rpm

Relates: https://github.com/389ds/389-ds-base/issues/4262

Reviewed by: viktor(Thanks!)

mreynolds389 committed:
Issue 4262 - Remove legacy tools subpackage (final cleanup)

Description:  Found a few more cleanup issues with removing perl.

Relates: https://github.com/389ds/389-ds-base/issues/4262

Reviewed by: spichugi(Thanks!)

Firstyear commented on ldap/servers/plugins/acl/acllist.c:
```diff
@@ -129,7 +129,7 @@ acl_be_state_change_fnc(void *handle __attribute__((unused)), char *be_name, int
         */
 
-        if ((sdn = slapi_be_getsuffix(be)) == NULL) {
+        if ((sdn = slapi_be_getsuffix(be, 0)) == NULL) {
```
I think there have been some changes around this, @mreynolds389 may know more. Either way, I don't think this change should be in this PR, I think that you may need to rebase to the latest commits.

Firstyear commented on ldap/servers/slapd/main.c:
```diff
@@ -538,6 +539,7 @@ main(int argc, char **argv)
             mallopt(M_MXFAST, val);
         }
     }
+#endif
```
What happens when we have linux but not glibc?

mreynolds389 committed:
Bump version to 1.4.4.6

kazimsarikaya committed:
Merge branch 'master' of https://github.com/kazimsarikaya/389-ds-base
